src/ui/component/stepper/useStepper.ts

```diff
@@ -0,0 +1,97 @@
+import { useCallback, useState } from 'react'
+import type { Step, StepElement, StepId, StepStatus } from './stepper'
+
+export type UseStepperInput = Array<Omit<StepElement, 'content'>>
+
+export type StepperControls = {
+  steps: Omit<Step, 'content'>[]
+  nextStep: () => void
+  previousStep: () => void
+  activeStepId: StepId
+  previousActiveStepId?: StepId
+}
+
+type ActiveStepsId = {
+  current: StepId
+  previous?: StepId
+}
```

Comment:
I can't really see why this is needed in the hook because you can always compute the previous active step based upon the active step, there is no need to memoize the `previous` active step through a state.

Moreover, `ActiveStepsId` suggests that there are several steps in memory.

Reply:
We can definitely improve the state naming. Regarding the previous active step, it's kept in memory because we don't have a way to figure which one was the previous one from the steps.